### ISSUE TITLE
inaccuracy with rendering UI using polygons

### ISSUE THREAD

**Issue by ToruTheRedFox** [gonetz/GLideN64#2390]
not sure how to really describe this, but the UVs seem to be off enough to shift textures on UI elements rendered using polygons and a 2D ortho matrix, which is especially noticeable at native resolution when the element isn't point filtered, making it really blurry

GLideN64: 
![image](https://user-images.githubusercontent.com/15708278/100128718-86f3a600-2e78-11eb-9336-530a3da3cf03.png)
AngryLion RDP:
![image](https://user-images.githubusercontent.com/15708278/100128944-d6d26d00-2e78-11eb-86ae-adda4f262e6f.png)

note how crisp the triangle in the middle of the map is compared to GLideN64, which is just a yellow smudge

(man, this map sure likes pointing out bugs in N64 emulation, doesn't it?)

**Comment by rez-** [comment]
maybe im totally wrong here but could this be caused by those filters like xrbz? altho i think xbrz shouldve actually made it clear on not blurry

**Comment by ToruTheRedFox** [comment]
I don't use any texture enhancement as it's counter-productive when developing

**Comment by ToruTheRedFox** [comment]
oh and likely related issue with MSAA on:

MSAA on:
![image](https://user-images.githubusercontent.com/15708278/100281463-de217580-2f61-11eb-94c2-f9e6f1401ed8.png)
MSAA off:
![image](https://user-images.githubusercontent.com/15708278/100281484-ea0d3780-2f61-11eb-8388-0ebc32803cbb.png)
AngryLion:
![image](https://user-images.githubusercontent.com/15708278/100281530-ff826180-2f61-11eb-885f-8c0a0075fb4e.png)

**Comment by ToruTheRedFox** [comment]
you guys sure are lucky Nintendo didn't go with texture wrapping for the power meter in the final game

**Comment by weinerschnitzel** [comment]
Is this a sm64 beta romhack, or the sm64 beta itself?

You mention "this map" and "development," for someone that doesn't know what you are working on, how would I repro?

**Comment by oddMLan** [comment]
> oh and likely related issue with MSAA on:
> 
> MSAA on:
> ![image](https://user-images.githubusercontent.com/15708278/100281463-de217580-2f61-11eb-94c2-f9e6f1401ed8.png)
> MSAA off:
> ![image](https://user-images.githubusercontent.com/15708278/100281484-ea0d3780-2f61-11eb-8388-0ebc32803cbb.png)
> AngryLion:
> ![image](https://user-images.githubusercontent.com/15708278/100281530-ff826180-2f61-11eb-885f-8c0a0075fb4e.png)

There's also a duplicate row of pixels right in the middle of GLideN64

**Comment by ToruTheRedFox** [comment]
that's mainly what I'm talking about

MSAA causes too much of the texture to be sampled at the edges

**Comment by iProgramMC** [comment]
> Is this a sm64 beta romhack, or the sm64 beta itself?
it's a rom hack

**Comment by iProgramMC** [comment]
related issue: the font looks slightly taller in SM64

GLideN64:
![image](https://user-images.githubusercontent.com/17717494/100726741-30a3cd00-337a-11eb-91ff-d90f17ca502b.png)

Angrylion:
![image](https://user-images.githubusercontent.com/17717494/100726798-444f3380-337a-11eb-817a-e383dda8c837.png)

**Comment by iProgramMC** [comment]
it seems like a UV mapping issue since you can simply replace two numbers to get it to look good on both

**Comment by oddMLan** [comment]
does it look the same if you try native resolution?

**Comment by iProgramMC** [comment]
nah, it gets slightly worse (half a pixel in 480p mode gets rounded to 1 pixel in 240p mode)

**Comment by ToruTheRedFox** [comment]
> > Is this a sm64 beta romhack, or the sm64 beta itself?
> 
> it's a rom hack

it uses the actual beta displaylist for the power meter though

**Comment by gonetz** [comment]
@ToruTheRedFox please check the issue with this build:
https://ci.appveyor.com/project/gonetz/gliden64/builds/38285888/artifacts

**Comment by ToruTheRedFox** [comment]
the minimap is definitely a lot less blurry now
![image](https://user-images.githubusercontent.com/15708278/112728164-c7922780-8f1d-11eb-9ff0-61fce40d6aa1.png)

**Comment by ToruTheRedFox** [comment]
(keep in mind that the texture has been updated since then)

**Comment by ToruTheRedFox** [comment]
the meter itself still is broken though 
![image](https://user-images.githubusercontent.com/15708278/112730102-8dc61e80-8f27-11eb-8425-0bc6e12e42ca.png)

**Comment by Jj0YzL5nvJ** [comment]
@ToruTheRedFox, can you point to the patch for that romhack? I can't find it anywhere.

**Comment by ToruTheRedFox** [comment]
it's not released yet @Jj0YzL5nvJ
